kv/kvserver/closedts/sidetransport: TestPaceUpdateSignalling failed

kv/kvserver/closedts/sidetransport.TestPaceUpdateSignalling [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21147058?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21147058?buildTab=artifacts#/) on release-26.1 @ [ca9f28b1fb4d3857777c4b19de5f595af9042df1](https://github.com/cockroachdb/cockroach/commits/ca9f28b1fb4d3857777c4b19de5f595af9042df1):


```
=== RUN   TestPaceUpdateSignalling
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/b281b3696cdd2199f40ec467f117c85a/logTestPaceUpdateSignalling2393111259
    test_log_scope.go:82: use -show-logs to present logs inline
    sender_test.go:1035: -- test log scope end --
--- FAIL: TestPaceUpdateSignalling (0.40s)
=== RUN   TestPaceUpdateSignalling/pacing_interval=0ms
    sender_test.go:1032: 
        	Error Trace:	pkg/kv/kvserver/closedts/sidetransport/sender_test.go:1032
        	            				pkg/kv/kvserver/closedts/sidetransport/sender_test.go:1005
        	            				pkg/kv/kvserver/closedts/sidetransport/sender_test.go:1029
        	Error:      	"32.306825ms" is not less than or equal to "30ms"
        	Test:       	TestPaceUpdateSignalling/pacing_interval=0ms
--- FAIL: TestPaceUpdateSignalling/pacing_interval=0ms (0.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPaceUpdateSignalling.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60700

--- a/pkg/kv/kvserver/closedts/sidetransport/sender_test.go
+++ b/pkg/kv/kvserver/closedts/sidetransport/sender_test.go
@@ -1034,9 +1034,10 @@ func TestPaceUpdateSignalling(t *testing.T) {
 	t.Run("pacing_interval=0ms", func(t *testing.T) {
 		closedts.SideTransportPacingRefreshInterval.Override(ctx, &st.SV, 0)
 		testPacing(t, 3 /* seqNum */, func(timeSpread time.Duration) {
-			// The expectation here is many 30x what is typically seen. But, we want
-			// to avoid flakes.
-			require.LessOrEqual(t, timeSpread, 30*time.Millisecond)
+			// With pacing disabled, all 1000 goroutines should be woken nearly
+			// simultaneously. The typical spread is ~1ms, but CI machines under
+			// load need more headroom for goroutine scheduling.
+			require.LessOrEqual(t, timeSpread, 40*time.Millisecond)
 		})
 	})
 }

